src/common/simple_spin.cc

```diff
@@ -31,9 +28,11 @@ void simple_spin_lock(simple_spinlock_t *lock)
 	return;
     }
     // delay
-    for (int i = 0; i < 100000; i++) {
-      *foo = (*foo * 33) + 17;
-    }
+#if defined(__arm__) || defined(__aarch64__)
+    asm volatile("yield");
+#else
```

Comment:
we need to check for at least `#if defined(__i386__)` here, AFAIK, we do build on ppc64. in that case, it would be 

```c
  asm volatile("or 27,27,27");
```

and fallback to the `for` loop if we are compiling on none of them or just emit an error/warning at compile time. IMHO, the latter is acceptable in practice.